compiler/rustc_expand/src/mbe/transcribe.rs

```diff
@@ -228,7 +228,7 @@ pub(super) fn transcribe<'a>(
             }
 
             // Replace the meta-var with the matched token tree from the invocation.
-            mbe::TokenTree::MetaVar(mut sp, mut original_ident) => {
+            mbe::TokenTree::MetaVar(mut sp, mut original_ident, undelimited_seq) => {
```

Comment:
I wonder if it's possible to detect variables from undelimited sequences like `$($var)*` right here in transcribe, instead of storing this property in `MetaVar` in advance.
(The transcribe code is tricky, and I didn't spend too much time on figuring this out.)
cc @mark-i-m @nnethercote maybe

Reply:
It is possible, I've updated the PR.